### ISSUE TITLE
When the program is doing a scan, there is a malfunction in the APIs

### ISSUE THREAD

**Issue by ErtyDess** [OWASP/Nettacker#762]
Please describe the issue or question and share your OS and Python version.
_________________
**OS**: `x`
Ubuntu
**OS Version**: `x`
22.04.2 LTS
**Python Version**: `x`
Python 3.10.12

Hello, i made a UI in react to call the apis.
When nettacker it's not scanning all the api works perfectly but when a scan is in progress the api responds almost every time "database error!" but every once in a while the api works.

**Comment by securestep9** [comment]
this looks like a multi-threading bug. When the scan is in progress the database is busy being constantly updated with scan results

**Comment by koffandaff** [comment]
Hello @securestep9 , I see that this issue is still open
 I am interested in working on this issue.

My analysis is that this is a classic concurrency bug caused by SQLite database locking in a high multi-threaded environment. When the scanner is writing constant results, it holds a lock, causing the API's read/write attempts (for session data, log lists, etc.) to frequently time out and return "database error!
I have experience debugging concurrency issues and would like to be assigned this task. Thank you

**Comment by securestep9** [comment]
hi @koffandaff have you actually managed to reproduce this issue? How are you proposing to fix it?

**Comment by koffandaff** [comment]
> hi [@koffandaff](https://github.com/koffandaff) have you actually managed to reproduce this issue? How are you proposing to fix it?

Yes @securestep9 , I've reproduced the issue consistently. The core problem is SQLite lock contention in the multi-threaded environment.

### **Findings:**

- Scan threads hold database locks for 3-10 seconds each
- API requests timeout after 5 seconds
- Results in "database error!" during active scans
- The error returns status 200 instead of 500 which makes it hard to identify

### **Scenario:**

When users run scans (e.g., 100+ concurrent threads checking targets), each thread writes results to the database. These writes create exclusive locks. Meanwhile, the React UI polls the API every 2 seconds to update progress. If an API read hits during a write lock, it waits 5 seconds (default SQLite timeout), then fails with "database error!" This happens constantly during active scans because write operations overlap significantly.

### **Error:**

<img width="1129" height="214" alt="Image" src="https://github.com/user-attachments/assets/9df92ae2-caa4-4942-b48d-1ab12cc9cccf" />

### **Proposed solutions (in priority order):**

- Enable SQLite WAL mode (biggest impact - allows concurrent reads/writes)
 
- Increase timeout from 5s to 30s (gives more time for locks to clear)

- Add retry logic for API reads (auto-retry on temporary locks)

- Fix HTTP 200/500 status codes (proper error signaling)

**Comment by securestep9** [comment]
Resolved in PR #1261
